test/example/scene_example.ex

```diff
@@ -0,0 +1,87 @@
+defmodule Example do
+  @moduledoc """
+  This module covers two examples for scene API.
+
+  An easy example simulates the layout of a video conferencing app.
+  There are 4 video inputs. All of them get their corners rounded. Then,
+  one of them is chosen as a main video (which takes most of the screen),
+  while the rest are scaled-down and put in a side strip on top of the main video.
+
+  A hard example takes three videos, applies varied transformations to them, and puts
+  them together on the final canvas
+  - The first video is simply rotated and then put in the top left corner
+  - The second video is merged with the first video. The result gets
+  corner rounding and is put in the top right corner
+  - The third video is turned into a ball and put in the middle bottom of the screen
+  """
+
+  alias Membrane.VideoCompositor.{Canvas, Compound, Scene, Texture}
+  alias Membrane.VideoCompositor.Compound.BaseSize
+  alias Mock.{CornerRounding, Grid, Merging, Overlay, Rotate, ToBall}
+
+  @easy %Scene{
+    alternations: [
+      rounding: %CornerRounding{pixels: 10, degrees: 60}
+    ],
+    layouts: [
+      overlay: Overlay,
+      strip: Grid
+    ],
+    objects: [
+      rounded_1: %Texture{input: :video_1, transformations: [:rounding]},
+      rounded_2: %Texture{input: :video_2, transformations: [:rounding]},
+      rounded_3: %Texture{input: :video_3, transformations: [:rounding]},
+      rounded_4: %Texture{input: :video_4, transformations: [:rounding]},
+      strip: %Compound{
+        base_size: %BaseSize{height: 1080, width: 480},
+        inputs_map: %{
+          rounded_1: :top,
+          rounded_2: :middle,
+          rounded_3: :bottom
+        },
+        layout: :strip
+      },
+      final_object: %Compound{
+        base_size: %BaseSize{height: 1080, width: 1920},
+        inputs_map: %{
+          strip: 1.0,
+          rounded_4: 0.0
+        },
+        layout: :overlay
+      }
+    ],
+    render: :final_object
+  }
```

Comment:
As you can see, `inputs_map` generates quite confusion. I wonder what is the reason for Compound to exist in the first place? IMO layouts would be enough and they could provide better APIs for defining inputs, e.g.

```elixir
%Overlay{
  base_size: %BaseSize{height: 1080, width: 1920},
  layers: [
    # position in lists determines the order of render, no need to add z-index
    # 2nd and 3rd item are x & y positions of input
    {:input1, 0, 0},
    {:input2, 0, 0}
  ]
}
%AdaptiveGrid{
  base_size: %BaseSize{height: 1080, width: 1920},
  inputs: [:input1, :input2, :input3, :input4]
}
# Grid with one main input and the rest grouped on the side
%PinnedItemGrid{
  base_size: %BaseSize{height: 1080, width: 1920},
  main: :input1,
  side: [:input2, :input3, :input4]
}
```